src/main/java/io/github/fixitlater/quizapi/question/QuestionService.java

```diff
@@ -67,6 +69,34 @@ public void addOne(QuestionDTO question) throws RuntimeException {
                 throw new UnableToSaveQuestionException("Unknown error occurred while saving new question");
             }
         }
+    }
+
+    public void deleteOne(Long questionId) {
+        questionRepository.deleteById(questionId);
+        if (findQuestionById(questionId).isPresent()) throw new DeleteUnsuccessfulException("Question was not deleted");
+    }
+
+    @Transactional()
+    public void updateOne(Long questionId, QuestionDTO questionDTO) {
+        QuestionEntity questionEntity = questionRepository.findById(questionId)
+                .orElseThrow(()-> new NoSuchElementException("Unable to modify question. Unable to find question with given ID"));
+        questionEntity.setQuestionBody(questionDTO.getQuestionBody());
+        questionEntity.setCategory(questionDTO.getCategory());
+        questionEntity.setLanguage(questionDTO.getLanguage());
 
+        if(questionDTO.getAnswers().size()!=questionEntity.getAnswersEntities().size()){
+            throw new UnableToSaveQuestionException("Number of answers cannot be changed during modification.");
+        }
+
+        for (int i = 0; i < questionEntity.getAnswersEntities().size(); i++){
+            questionEntity.getAnswersEntities().get(i).setAnswerBody(questionDTO.getAnswers().get(i).getAnswerBody());
+            questionEntity.getAnswersEntities().get(i).setCorrect(questionDTO.getAnswers().get(i).isCorrect());
+        }
+        try {
+            questionRepository.save(questionEntity);
+        } catch (Exception e) {
```

Comment:
Ten try z ogólnym Exception jest nieco zaskakujący w tym miejscu.
Takich ogólnych exceptionów, lepiej nie łapać, w szczególności, gdy nie przekazujemy dalej stacktrace, bo będzie nam trudniej wykryć potencjalne problemy, jeżeli wystąpią.
Ja całokiwicie bym się tego pozbył, a jeżeli zostało to zrobione, z powodu, że czasem lecą jakieś wyjątki, to lepiej naprawić przyczynę ;-)